lib/rubocop/cop/rails/order_arguments.rb

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # Prefer symbol arguments over strings in `order` method.
+      #
+      # @safety
+      #   Cop is unsafe because the receiver might not be an Active Record query.
+      #
+      # @example
+      #   # bad
+      #   User.order('name')
+      #   User.order('name DESC')
+      #
+      #   # good
+      #   User.order(:name)
+      #   User.order(name: :desc)
+      #
+      class OrderArguments < Base
+        extend AutoCorrector
+
+        MSG = 'Prefer `%<prefer>s` instead.'
+
+        RESTRICT_ON_SEND = %i[order].freeze
+
+        def_node_matcher :string_order, <<~PATTERN
+          (call _ :order (str $_value)+)
+        PATTERN
+
+        SORT_EXPRESSION_REGEX = /^(\w+) ?(asc|desc)?$/i.freeze
+
+        def on_send(node)
+          return unless (sort_expressions = string_order(node))
+          return unless (new_order = replacement(sort_expressions))
+
+          offense_range = find_offense_range(node)
+          add_offense(offense_range, message: format(MSG, prefer: new_order)) do |corrector|
+            corrector.replace(offense_range, new_order)
+          end
+        end
+        alias on_csend on_send
+
+        private
+
+        def find_offense_range(node)
+          node.first_argument.source_range.join(node.last_argument.source_range)
+        end
+
+        def replacement(sort_expressions)
+          sort_expressions = sort_expressions.flat_map { _1.split(',') }.map(&:strip).map { column_and_dir(_1) }
+
+          return if sort_expressions.any?(&:nil?)
+
+          to_arguments(sort_expressions).join(', ')
+        end
+
+        def to_arguments(sort_expressions)
+          use_hash = false
+          sort_expressions.map do |column, direction|
+            if direction == :asc && !use_hash
+              ":#{column}"
+            else
+              use_hash = true
+              "#{column}: :#{direction}"
+            end
+          end
+        end
```

Comment:
Using `order` instead of `sort` would make the context easier to follow.

```suggestion
        def convert_to_preferred_arguments(order_expressions)
          use_hash = false
          order_expressions.map do |column, direction|
            if direction == :asc && !use_hash
              ":#{column}"
            else
              use_hash = true
              "#{column}: :#{direction}"
            end
          end
        end
```